### PR TITLE
Blaze: Create campaign from the payment screen

### LINKED ISSUE #11702: Blaze: Payment screen
image

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #11702 

## Description

This PR integrates campaign creation on the payment screen when a selected payment method is available.

The loading state has been handled on the UI, and the callback closure is triggered upon success. The success bottom sheet and error state should be handled in separate PRs.

## Testing instructions

- Log in to a store that's eligible for Blaze.
- If the campaign creation entry point is available on the Dashboard screen, select it and select a product if needed. Otherwise, navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- On the payment screen, tap Submit Campaign.
- Confirm that a loading screen is displayed briefly.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/5533851/545ecc50-7b90-489c-b447-2f267c54ef15

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift [hunk 1]
```diff
@@ -47,10 +54,29 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
...
         } catch {
             DDLogError("⛔️ Error fetching payment info for Blaze campaign creation: \(error)")
-            shouldDisplayErrorAlert = true
+            shouldDisplayPaymentErrorAlert = true
         }
         isFetchingPaymentInfo = false
     }
+
+    @MainActor
+    func confirmPaymentDetails() async {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]: Opinion: What do you think about renaming this into `createCampaign` as this method actually starts the campaign creation process?

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift [hunk 2]
```diff
@@ -47,10 +54,29 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
...
+        isCreatingCampaign = true
+        do {
+            let updatedDetails = campaignInfo.copy(paymentMethodID: selectedPaymentMethod.id)
+            try await requestCampaignCreation(details: updatedDetails)
+        } catch {
+            DDLogError("⛔️ Error creating Blaze campaign: \(error)")
+            shouldDisplayCampaignCreationError = true
+        }
+        isCreatingCampaign = false
+        completionHandler()
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]: Nit: Should we call `completionHandler` only upon success?

**Review by selanthiraiyan — APPROVED** [review]
:shipit:

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]: Yes, this was a mistake 😅 Updated in 3a07b222f9d96bc11e572a9ba8584d40f7b465dd.

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift]: I renamed this to `submitCampaign` to match the CTA in a30baf248c3933865772b3dd1dbaad7c9512df6e.

**Pull request merged**